src/dhtproto/node/neo/request/GetAll.d

```diff
@@ -56,6 +56,61 @@ public abstract scope class GetAllProtocol_v0
     /// If true, only record keys will be sent, no values.
     private bool keys_only;
 
+    /// Value filtering wrapper struct.
+    private struct ValueFilter
+    {
+        import ocean.text.Search;
+
+        /// Sub-array matcher. (Type ubyte as doesn't compile with void.)
+        private SearchFruct!(Const!(ubyte)) matcher;
+
+        /// Filtering active?
+        private bool active;
+
+        /***********************************************************************
+
+            Initialises the filter from the specified filter array.
+
+            Params:
+                filter = sub-array to filter records by. If empty, no filtering
+                    is employed
+
+        ***********************************************************************/
+
+        public void init ( in void[] filter )
+        {
+            if ( filter.length > 0 )
+            {
+                this.matcher = search(cast(Const!(ubyte)[])filter);
+                this.active = true;
+            }
+        }
+
+        /***********************************************************************
+
+            Checks whether the provided record value passes the filter or not.
+
+            Params:
+                value = value to filter
+
+            Returns:
+                true if the value passes (i.e. matches) the filter or if
+                filtering is not active
+
+        ***********************************************************************/
+
+        public bool match ( in void[] value )
+        {
+            if ( this.active )
```

Comment:
`assert (this.matcher !is null)`

Reply:
I think this is impossible. `SearchFruct!(T)` is a struct and the internal "has this been setup?" fields are private.

Reply:
:+1: